client/network/src/protocol.rs

```diff
@@ -1157,24 +1166,37 @@ impl<B: BlockT, H: ExHashT> Protocol<B, H> {
 		validation_result: sync::PollBlockAnnounceValidation<B::Header>,
 	) -> CustomMessageOutcome<B> {
 		let (header, is_best, who) = match validation_result {
-			sync::PollBlockAnnounceValidation::Nothing { is_best, who, header } => {
+			sync::PollBlockAnnounceValidation::Nothing { is_best, who, announce } => {
 				self.update_peer_info(&who);
 
+				if let Some(data) = announce.data {
+					if !data.is_empty() {
+						self.block_announce_data_cache.put(announce.header.hash(), data);
```

Comment:
In this case should we overwrite any existing block announce data that might already exist? My understanding is that we might not actually validate the block announce data when we get `PollBlockAnnounceValidation::Nothing`.

Reply:
Good point. Removed it.

Reply:
Okay readded it xD But changed the logic a little bit :P